app/utils_k8s.py

```diff
@@ -40,3 +45,32 @@ def k8s_read_namespaced_deployment(
         if ex.status == 404:
             return None
         raise
+
+
+def k8s_get_secret(namespace: str, name: str) -> kubernetes.client.V1Secret | None:
```

Comment:
Other methods in this module also get the `kapi: kubernetes.client.CoreV1Api | None = None` kwarg and only instanciate a new `CoreV1API` if not given one.
Lets keep the same behavior here?